__tests__/installer.test.ts

```diff
@@ -549,6 +557,108 @@ describe('setup-node', () => {
     });
   });
 
+  describe('node-version-file flag', () => {
+    it('not used if node-version is provided', async () => {
+      // Arrange
+      inputs['node-version'] = '12';
+
+      // Act
+      await main.run();
+
+      // Assert
+      expect(parseNodeVersionSpy).toHaveBeenCalledTimes(0);
+    });
+
+    it('not used if node-version-file not provided', async () => {
+      // Act
+      await main.run();
+
+      // Assert
+      expect(parseNodeVersionSpy).toHaveBeenCalledTimes(0);
+    });
+
+    it('warns if node-version and node-version-file are provided', async () => {
+      //Arrange
+      inputs['node-version'] = '12';
+      inputs['node-version-file'] = '__tests__/data/.nvmrc';
+      // Act
+      await main.run();
+
+      // Assert
+      expect(warningSpy).toHaveBeenCalledTimes(1);
+      expect(warningSpy).toHaveBeenCalledWith(
+        `Both node-version and node-version-file inputs are specified, only node-version will be used`
+      );
+    });
+
+    it('reads node-version-file if provided', async () => {
+      // Arrange
+      const versionSpec = 'v14';
+      const versionFile = '.nvmrc';
+      const expectedVersionSpec = '14';
+      process.env['GITHUB_WORKSPACE'] = path.join(__dirname, 'data');
+      inputs['node-version-file'] = versionFile;
+
+      parseNodeVersionSpy.mockImplementation(() => expectedVersionSpec);
+      existsSpy.mockImplementationOnce(
+        input => input === path.join(__dirname, 'data', versionFile)
+      );
+      // Act
+      await main.run();
+
+      // Assert
+      expect(existsSpy).toHaveBeenCalledTimes(1);
+      expect(existsSpy).toHaveReturnedWith(true);
+      expect(parseNodeVersionSpy).toHaveBeenCalledWith(versionSpec);
+      expect(logSpy).toHaveBeenCalledWith(
+        `Resolved ${versionFile} as ${expectedVersionSpec}`
+      );
+    });
+
+    it('both node-version-file and node-version are provided', async () => {
```

Comment:
This test case is similar to `warns if node-version and node-version-file are provided`. I think one of them should be removed